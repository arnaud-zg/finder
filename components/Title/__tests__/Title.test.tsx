import React from 'react'
import renderer from 'react-test-renderer'
import { Title } from '../'

describe('Title', () => {
  it('renders null', () => {
    expect(
      renderer.create(<Title content="" level={0} />).toJSON()
    ).toMatchSnapshot()
  })

  it('renders correctly with property level 1', () => {
    expect(
      renderer.create(<Title content="Hello World" level={1} />).toJSON()
    ).toMatchSnapshot()

    expect(
      renderer
        .create(<Title center={true} content="Hello World" level={1} />)
        .toJSON()
    ).toMatchSnapshot()
  })

  it('renders correctly with property level 2', () => {
    expect(
      renderer.create(<Title content="Hello World" level={2} />).toJSON()
    ).toMatchSnapshot()

    expect(
      renderer
        .create(<Title center={true} content="Hello World" level={2} />)
        .toJSON()
    ).toMatchSnapshot()
  })

  it('renders correctly with property level 3', () => {
    expect(
      renderer.create(<Title content="Hello World" level={3} />).toJSON()
    ).toMatchSnapshot()

    expect(
      renderer
        .create(<Title center={true} content="Hello World" level={3} />)
        .toJSON()
    ).toMatchSnapshot()
  })
})
