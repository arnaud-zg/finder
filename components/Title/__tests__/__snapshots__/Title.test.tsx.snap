// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders correctly with property level 1 1`] = `
<h1
  className="f9o1qoo"
>
  Hello World
</h1>
`;

exports[`Title renders correctly with property level 1 2`] = `
<h1
  className="f9o1qoo f1jc9aff"
>
  Hello World
</h1>
`;

exports[`Title renders correctly with property level 2 1`] = `
<h2
  className="f113tx4b"
>
  Hello World
</h2>
`;

exports[`Title renders correctly with property level 2 2`] = `
<h2
  className="f113tx4b f1jc9aff"
>
  Hello World
</h2>
`;

exports[`Title renders correctly with property level 3 1`] = `
<h3
  className="fhwo3dt"
>
  Hello World
</h3>
`;

exports[`Title renders correctly with property level 3 2`] = `
<h3
  className="fhwo3dt f1jc9aff"
>
  Hello World
</h3>
`;

exports[`Title renders null 1`] = `null`;
